src/xERC20/contracts/XUltraLRT.sol

```diff
@@ -152,11 +152,11 @@ contract XUltraLRT is
      * @param receiver The address of the receiver
      */
     function deposit(uint256 _amount, address receiver) public whenNotPaused onlyTokenDepositAllowed {
-        require(block.timestamp - lastPriceUpdateTimeStamp <= maxPriceLag, "XUltraLRT: Price not updated");
-        require(_amount > 0, "XUltraLRT: Invalid amount");
-        require(receiver != address(0), "XUltraLRT: Invalid receiver");
-        require(sharePrice > 0, "XUltraLRT: Invalid share price");
-        require(address(baseAsset) != address(0), "XUltraLRT: Invalid base asset");
+        if (block.timestamp - lastPriceUpdateTimeStamp > maxPriceLag) revert XErrors.NotUpdatedPrice();
+        if (_amount == 0) revert XErrors.InvalidAmount();
+        if (receiver == address(0)) revert XErrors.InvalidReceiver();
+        if (sharePrice == 0) revert XErrors.InvalidSharePrice();
```

Comment:
also do we really need to do all these checks inside every single deposit?

Reply:
Yup, we can't store log.